aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -295,6 +299,10 @@ module aptos_framework::delegation_pool {
         delegate_voting_power_events: EventHandle<DelegateVotingPowerEvent>,
     }
 
+    struct Beneficiary has key {
```

Comment:
Would the following design be an option worth exploring?
- `Beneficiary` is stored on the resource account of the delegation pool
- Operator which is the default/initial beneficiary can set the next beneficiary the first time
- Only the current beneficiary can change themselves to a different account
Usually, the operator account is used for operations on the `stake` module to rotate validator keys in case of infrastructure migrations etc. The entity managing the validator software usually grants access to the operator to their infrastructure/DevOps team. The beneficiary can be a cold wallet owned by the finance department, but in the current design the operator can still change it which is not completely separating:

- who manages operator-oriented operations
- who receives the rewards for restaking/withdrawing them

Reply:
The main complexity this would lead to is dealing with wiping the beneficiary account when changing operators

Reply:
yes, changing the operator requires defaulting the beneficiary too (e.g. to this new operator's address)

Reply:
The permission structure of the current design is: the owner has permission to set an operator, and the operator has permission to set a beneficiary. I believe that the current design satisfies the requirements that @michelle-aptos has. @michelle-aptos, do you have any comments on this?

Reply:
Also, if the operator has permission to change its beneficiary, it can handle the situation where the beneficiary account is compromised.

Reply:
the operator should only have permission to change the beneficiary, but the beneficiary receives the rewards and manages the restaking/withdrawal.

Reply:
agree with the original comment the beneficiary should be controlled by validator...

Reply:
This feature is for the operator rather than the pool owner. This feature allows the operator to choose where to receive the operator commission.